prov/rxm/src/rxm.h

```diff
@@ -1033,22 +1042,37 @@ rxm_tx_buf_get(struct rxm_ep *rxm_ep, enum rxm_buf_pool_type type)
 	       (type == RXM_BUF_POOL_TX_ACK) ||
 	       (type == RXM_BUF_POOL_TX_RNDV) ||
 	       (type == RXM_BUF_POOL_TX_SAR));
-	return (struct rxm_tx_buf *)rxm_buf_get(&rxm_ep->buf_pools[type]);
+	return (union rxm_tx_buf *)rxm_buf_get(&rxm_ep->buf_pools[type]);
 }
 
 static inline void
-rxm_tx_buf_release(struct rxm_ep *rxm_ep, struct rxm_tx_buf *tx_buf)
+rxm_tx_buf_release(struct rxm_ep *rxm_ep, void *tx_buf)
 {
-	assert((tx_buf->type == RXM_BUF_POOL_TX) ||
-	       (tx_buf->type == RXM_BUF_POOL_TX_INJECT) ||
-	       (tx_buf->type == RXM_BUF_POOL_TX_ACK) ||
-	       (tx_buf->type == RXM_BUF_POOL_TX_RNDV) ||
-	       (tx_buf->type == RXM_BUF_POOL_TX_SAR));
-	/*assert((tx_buf->pkt.ctrl_hdr.type == ofi_ctrl_data) ||
-	       (tx_buf->pkt.ctrl_hdr.type == ofi_ctrl_large_data) ||
-	       (tx_buf->pkt.ctrl_hdr.type == ofi_ctrl_seg_data) ||
-	       (tx_buf->pkt.ctrl_hdr.type == ofi_ctrl_ack));*/
-	rxm_buf_release(&rxm_ep->buf_pools[tx_buf->type],
+#ifdef NDEBUG
```

Comment:
Did you mean ENABLE_DEBUG?

Reply:
I think we can use `NDEBUG` here, because this just does `assert()`

Reply:
Removed this :)